rand_core/src/impls.rs

```diff
@@ -109,4 +110,39 @@ pub fn next_u128_via_fill<R: Rng+?Sized>(rng: &mut R) -> u128 {
     impl_uint_from_fill!(rng, u128, 16)
 }
 
+const LIMIT_ERR_MAX: usize = 20;    // arbitrary
+static LIMIT_ERR_COUNT: AtomicUsize = ATOMIC_USIZE_INIT;
+
+/// Implement `fill_bytes` via `try_fill` with implicit error handling.
+pub fn fill_via_try_fill<R: Rng+?Sized>(rng: &mut R, dest: &mut [u8]) {
+    loop {
+        if let Err(e) = rng.try_fill(dest) {
+            if e.kind.should_retry() {
+                if e.kind.limit_retries() {
+                    // We use a global counter since we don't have any local memory.
```

Comment:
I don't really get why this should be a global counter. Can't we just do a couple of retries here?